In [1]:
import nltk
nltk.download('vader_lexicon')
from nltk.sentiment import SentimentIntensityAnalyzer

import numpy as np

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     C:\Users\Abdullah\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


In [2]:
analyzer = SentimentIntensityAnalyzer()


## We start the analysis for each student

### Student 1

In [3]:
st1txt = "This was very knowledge able chapter with very well explained content"

In [4]:
scores1 = analyzer.polarity_scores(st1txt)

In [5]:
print(scores1)

{'neg': 0.0, 'neu': 0.807, 'pos': 0.193, 'compound': 0.3384}


## Student 2

In [9]:
st2txt = "NAN"

In [10]:
scores2 = analyzer.polarity_scores(st2txt)

In [11]:
print(scores2)

{'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}


## Student 3

In [13]:
st3txt = "The explanation of moral decision action and Omission is quite helpful. The concept of accountability is clear and the role of the individual stakeholder in developing the healthcare system is also clear. There is no difficulty understanding the chapter's content."

In [14]:
scores3 = analyzer.polarity_scores(st3txt)

In [15]:
print(scores3)

{'neg': 0.096, 'neu': 0.731, 'pos': 0.173, 'compound': 0.5709}


## Student 4

In [16]:
st4txt = "Accountability is an interesting topic, particularly, when considering autonomous AI systems. Therefore, I find this chapter interesting and worthy of discussion. I would also see some real use cases where there was an issue with accountability."

In [17]:
scores4 = analyzer.polarity_scores(st4txt)

In [18]:
print(scores4)

{'neg': 0.0, 'neu': 0.789, 'pos': 0.211, 'compound': 0.8074}


## Student 5

In [19]:
st5txt = "The tasks were clearly formulated. However, I was not absolutely sure what answers were desired."

In [20]:
scores5 = analyzer.polarity_scores(st5txt)

In [21]:
print(scores5)

{'neg': 0.121, 'neu': 0.612, 'pos': 0.267, 'compound': 0.3861}


## Student 6

In [53]:
st6txt = "Usefulness: As has been my feedback to previous chapters, I think the course should focus on the more foundational/philosophical concerns and try to incite lively discussions among the students. Such high level (high in abstraction hierarchy) description can only serve the purpose of creating an illusion of having learnt something.<br /> Clarity: My answers to the questions should serve as the feedback to this part.<br /> Difficulty: Not hard to follow. Any difficulty there was, arose out of lack of clarity."

In [54]:
scores6 = analyzer.polarity_scores(st6txt)

In [55]:
print(scores6)

{'neg': 0.077, 'neu': 0.771, 'pos': 0.152, 'compound': 0.7092}


## Student 7

In [56]:
st7txt = "This chapter explains very well."

In [57]:
scores7 = analyzer.polarity_scores(st7txt)

In [58]:
print(scores7)

{'neg': 0.0, 'neu': 0.626, 'pos': 0.374, 'compound': 0.3384}


## Student 8

In [59]:
st8txt = "This chapter was informative in providing a real-world scenario of the ethical dilemmas that can arise with the implementation of AI systems. It was useful to understand the various responsibilities of those involved in the deployment of such systems. In terms of clarity, the chapter was easy to follow and understand. I had no difficulty providing answers and comprehending the content."

In [60]:
scores8 = analyzer.polarity_scores(st8txt)

In [61]:
print(scores8)

{'neg': 0.066, 'neu': 0.764, 'pos': 0.17, 'compound': 0.802}


## Student 9

In [62]:
st9txt = "Some of the questions are biased, for example I would never consider a private company to develop a healthcare system that sensible, yet that was not to discuss. <br /> <br /> <br /> Also you made Ada Lovelace ( I assume that was the reference) in the first question a male character, which is quite sad..."

In [63]:
scores9 = analyzer.polarity_scores(st9txt)

In [64]:
print(scores9)

{'neg': 0.04, 'neu': 0.96, 'pos': 0.0, 'compound': -0.2732}


## Student 10

In [65]:
st10txt = "The chapter was clear and easy to go through."

In [66]:
scores10 = analyzer.polarity_scores(st10txt)

In [67]:
print(scores10)

{'neg': 0.0, 'neu': 0.56, 'pos': 0.44, 'compound': 0.6705}


## Student 11

In [68]:
st11txt = "It was useful in a philosophical study, and also in regard as to recapitulate / combine with AI-topics.<br /> It was mostly clear, but I often skipped back to the slide with the explanation text to re-read the attributes that were important, maybe you can summarize them on the other slides with the tasks.<br /> It was not so difficult and some-how time-consuming"

In [69]:
scores11 = analyzer.polarity_scores(st11txt)

In [70]:
print(scores11)

{'neg': 0.0, 'neu': 0.856, 'pos': 0.144, 'compound': 0.8145}


## Student 12

In [71]:
st12txt = "This chapter is designed with some real-life examples that help for better understanding."

In [72]:
scores12 = analyzer.polarity_scores(st12txt)

In [73]:
print(scores12)

{'neg': 0.0, 'neu': 0.663, 'pos': 0.337, 'compound': 0.6808}


## Student 13

In [74]:
st13txt = "The chapter, exposed possible ethical issues in more detail with a number of examples."

In [75]:
scores13 = analyzer.polarity_scores(st13txt)

In [76]:
print(scores13)

{'neg': 0.08, 'neu': 0.619, 'pos': 0.301, 'compound': 0.552}


## Student 14

In [77]:
st14txt = "feedback is sadly already contained in the answers,"

In [78]:
scores14 = analyzer.polarity_scores(st14txt)

In [79]:
print(scores14)

{'neg': 0.286, 'neu': 0.714, 'pos': 0.0, 'compound': -0.4215}


## Student 15

In [80]:
st15txt = "Generally, I think accountability is usually considered from the who did what perspective and it is great that this chapter prompts us on the fact that people who also stand by and watch actions occur are also accountable."

In [81]:
scores15 = analyzer.polarity_scores(st15txt)

In [82]:
print(scores15)

{'neg': 0.0, 'neu': 0.898, 'pos': 0.102, 'compound': 0.6249}


## Student 16

In [83]:
st16txt = "Ilias still shows an error which confuses me a bit."

In [84]:
scores16 = analyzer.polarity_scores(st16txt)

In [85]:
print(scores16)

{'neg': 0.417, 'neu': 0.583, 'pos': 0.0, 'compound': -0.6124}


In [86]:
# create a list of dictionaries to add
scores_list = [scores1, scores2, scores3, scores4, scores5, scores6, scores7, scores8,
               scores9, scores10, scores11, scores12, scores13, scores14, scores15, scores16]

# specify the key i need, in this case, compound
key = 'compound'

# create a list to store the values of each key that i will extract from each dict
values_list = []

# loop over the list of dictionaries and extract the values for the specific key
for d in scores_list:
    if key in d:
        values_list.append(d[key])

# calculate the average of the values:
average = np.mean(values_list)

print(average)

0.37424999999999997


# Week 3/ Chapter 3 scores are:

### The average score for week one sentiment analysis is 0.374 (1 being the most positive, 0 neutral, and -1 being negative)